### PR TITLE
Add Job "POST User Exports" endpoint

### PR DESCRIPTION
**Pull request by lbalmaceda**
### Changes

Adds the following endpoint:
- POST Users Export Job (by connection id)

### References

Will close https://github.com/auth0/auth0-java/issues/95

### Testing

- [x] This change adds test coverage
- [x] This change has been tested on the latest version of the platform/language or why not

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)
- [ ] All existing and new tests complete without errors

### CODE REVIEW

**Comment by lbalmaceda** [comment]
Yes, my intention was to add them to this PR. Let me mark it as in progress.

File: src/main/java/com/auth0/client/mgmt/JobsEntity.java [hunk 1]
```diff
@@ -50,4 +72,35 @@
...
+    /**
+     * Requests a Users Exports job. A token with scope read:users is needed.
+     * See https://auth0.com/docs/api/management/v2#!/Jobs/post_users_exports
+     * See https://auth0.com/docs/users/guides/bulk-user-exports
+     *
+     * @param connectionId The id of the connection to export the users from
+     * @param filter       the filter to use. Can be null.
+     * @return a Request to execute.
+     */
+    public Request<Job> requestUsersExport(String connectionId, UsersExportFilter filter) {
```
Comment (lbalmaceda) [inline src/main/java/com/auth0/client/mgmt/JobsEntity.java:85]: Open to name suggestions: `requestUsersExport`, `startUsersExport`, `beginUsersExport`, `postUsersExport`.

File: src/main/java/com/auth0/json/mgmt/jobs/UsersExportField.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class UsersExportField {
+
+    @JsonProperty("name")
+    private String name;
+    @JsonProperty("export_as")
+    private String exportAs;
+
+    public UsersExportField(String name) {
```
Comment (lbalmaceda) [inline src/main/java/com/auth0/json/mgmt/jobs/UsersExportField.java:20]: Keeping this one handy as seems to be the default scenario. If the alternative constructor is not required, I can replace it with a setter for `export_as`

File: src/test/java/com/auth0/client/MockServer.java [hunk 3]
```diff
@@ -117,7 +119,7 @@ public void jsonResponse(String path, int statusCode) throws IOException {
         server.enqueue(response);
     }
 
-    public void rateLimitReachedResponse(long limit, long remaining, long reset) throws IOException {
```
Comment (lbalmaceda) [inline src/test/java/com/auth0/client/MockServer.java:120]: linter pointed out exceptions not thrown, and auto-closing of buffer (with jdk 8 try/catch)

File: src/main/java/com/auth0/client/mgmt/JobsEntity.java (hunk 1 above)
Comment (lbalmaceda) [reply src/main/java/com/auth0/client/mgmt/JobsEntity.java:85]: @jimmyjames @damieng Also, is this supposed to be...? 

a) "userS exportS" 
b) "user exportS"
c) "users export" (the one I chose)

**Comment by lbalmaceda** [comment]
I'm working on adding the import users endpoint to this PR.

File: src/main/java/com/auth0/client/mgmt/JobsEntity.java (hunk 1 above)
Comment (damieng) [reply src/main/java/com/auth0/client/mgmt/JobsEntity.java:85]: I mean I think it should be `exportUsers` ...

**Comment by lbalmaceda** [comment]
I'm going to keep this one for the User Exports endpoint and add the remaining endpoints in separate PRs.

File: src/main/java/com/auth0/client/mgmt/JobsEntity.java (hunk 1 above)
Comment (lbalmaceda) [reply src/main/java/com/auth0/client/mgmt/JobsEntity.java]: Updated!

**Pull request merged**